### PR TITLE
To dataframe

### PR DESCRIPTION
**Pull request by romainmartinez**
## Proposed changes

Pandas dataframes are very efficient and are the standard for many visualisations, statistics and machine learning librairies.
Therefore, I added a `to_dataframe` method, which plays nicely with the `update_misc` method:

```python
DATA_PATH = Path('your_data_path')
emgs_list = pd.concat(
    [
        Analogs3d.from_c3d(ifile, names=["delt_ant"])
        .update_misc({"group": "female", "filename": ifile.stem})
        .to_dataframe(add_misc=True)
        for ifile in DATA_PATH.glob("*.c3d")
    ]
)
```

Which outputs a single dataframe for all of the file present in the `DATA_PATH` dir:
```
      delt_ant   group   filename
0    -0.258042  female  12kg_H2_3
1     0.494952  female  12kg_H2_3
2     0.423346  female  12kg_H2_3
3     0.446372  female  12kg_H2_3
4     0.480231  female  12kg_H2_3
5     0.397767  female  12kg_H2_3
```

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: pyomeca/frame_dependent.py [hunk 1]
```diff
@@ -307,6 +308,24 @@ def get_specific_data(self, idx):
...
     # --- Fileio methods (to_*)
 
+    def to_dataframe(self, add_misc=False):
+        """
+        Convert a Vectors3d class to a pandas dataframe
+
+        Parameters
+        ----------
+        add_misc : bool
+            add columns for each of the `misc` fields added with `self.update_misc` if True
```
[the remark this answers is not in the document]
Comment (romainmartinez) [reply pyomeca/frame_dependent.py]: I agree that we should support the addition of any metadata. So I replaced `add_misc` with a `add_metadata` list in my last commit.

you can now do:
```python
( 
    Analogs3d.from_c3d("c.c3d") 
    .update_misc({"group": "female"}) 
    .to_dataframe(add_metadata=["get_rate", "misc"]) 
) 
```

On the other hand, I don't think we should return all metadata by default, as in 95% of cases we won't use them. The size of the dataframe being consequent (rows: number of attempts times number of frames), the memory use of the dataframe would explode. 
And I like the fact that I can add random metadata (group, condition, etc.) so I would keep `misc`.

File: pyomeca/frame_dependent.py [hunk 2]
```diff
@@ -97,6 +97,7 @@ def update_misc(self, d):
             Dictionary to be added to the misc field
         """
         self.misc.update(d)
+        return self.dynamic_child_cast(self)
```
[the remark this answers is not in the document]
Comment (pariterre) [reply pyomeca/frame_dependent.py]: I see, maybe add in the description that an optional reference to the internal state is also return in order to chain the operation if needed?

[the remark this answers is not in the document]
Comment (romainmartinez) [reply pyomeca/frame_dependent.py]: Done :)

[the remark this answers is not in the document]
Comment (romainmartinez) [reply pyomeca/frame_dependent.py]: Done :)

**Pull request merged**